src/main/java/redis/clients/util/Pool.java

```diff
@@ -48,6 +49,11 @@ public T getResource() {
     try {
       return internalPool.borrowObject();
     } catch (NoSuchElementException nse) {
+      if (null == nse.getCause()) { // The exception was caused by an exhausted pool
```

Comment:
@shmulik-ubimo got it. If you've been looking into the commos-pool code then it's fine. I believe it might be better then to add this logic to JedisPool instead of just JedisCluster. What do you think?. This way if someone is using `JedisPool` it will get this new exception